### ISSUE TITLE
[WIP] tf 2.0 implementation of tts

### ISSUE THREAD

**Issue by erogol** [mozilla/TTS#300]
With the arrival of tf 2.0, I like to give it a try. Probably it would be a separate repo but might be useful for deployment. I'd guess it'd be ready in a week.

**Comment by jinfagang** [comment]
@erogol  How's the work going bro.

**Comment by erogol** [comment]
https://github.com/erogol/TTS_tf

**Comment by erogol** [comment]
This is pretty much stuck with TF 2.0 deficiencies especially with changing sequence length causing constant retracing.

**Comment by carlfm01** [comment]
@erogol  what about https://www.tensorflow.org/guide/ragged_tensor?  Were introduced to deal with dimension changing tensor, unfortunately as far as I know there's no benchmark

**Comment by erogol** [comment]
I don't think I have enough resources right now to work on this. If anyone is volunteering, you are more then welcome to give it a go. I am also ready to help.

**Comment by monatis** [comment]
hi @erogol, with [tf.text](https://www.tensorflow.org/tutorials/tensorflow_text/intro) and its preprocessing layers returning `RaggedTensor`, it's much easier to work with inputs of different lengths. Now that I'm working on a neural TTS project in Turkish (novel dataset, [sound annotating utility](https://github.com/monatis/label-snd) and training code), I'm volunteer to give it a try. What should I do next? Forking the repo, or any other model for collaboration?

**Comment by erogol** [comment]
You can fork it and give it a go. Then we can iterate together.

**Comment by monatis** [comment]
Great, let you know when I make a progress.

**Comment by erogol** [comment]
In case anyone is interested.  I reiterate TTS with Tensorflow eager mode.

@monatis any updates on your side.

**Comment by erogol** [comment]
> 
> 
> Hello @erogol, what about https://www.tensorflow.org/guide/ragged_tensor? Were introduced to deal with dimension changing tensor, unfortunately as far as I know there's no benchmark

Ragged Tensor looks promising but I am still struggling to convert TTS to TTS_tf due to other TF specific problems.
Such as, in eager mode tf.function tracing, I could not figure out the way to dynamically generate new tensors depending on the input shape to a specific module  [here](https://github.com/erogol/TTS_tf/blob/master/layers/common_layers.py#L106) since input tensor has a partial shape.

**Comment by erogol** [comment]
I reopen and close this as I added TF 2.0 Tacotron2 implementation.
